src/views/shared/snackbar/snackbar.view.tsx

```diff
@@ -1,33 +1,71 @@
 import React from "react";
 
 import Container from "src/views/shared/container/container.view";
+import Button from "src/views/shared/button/button.view";
+import { ReactComponent as CloseIconDark } from "src/images/icons/close.svg";
+import { ReactComponent as CloseIconLight } from "src/images/icons/close-white.svg";
 import useSnackbarStyles from "src/views/shared/snackbar/snackbar.styles";
 import { SNACKBAR_AUTO_HIDE_DURATION } from "src/constants";
+import theme from "src/styles/theme";
+//domain
+import { Message } from "src/domain";
 
 interface SnackbarProps {
-  message: string;
+  message: Message;
   backgroundColor?: string;
   onClose: () => void;
+  onReport: (raw: unknown, parsed: string) => void;
 }
 
-function Snackbar({ message, backgroundColor, onClose }: SnackbarProps): JSX.Element {
-  const classes = useSnackbarStyles({ backgroundColor });
+function Snackbar({ message, backgroundColor, onClose, onReport }: SnackbarProps): JSX.Element {
+  const bkgColor = backgroundColor
```

Comment:
Importing the `theme` directly from the `src/styles/theme` is not a good idea. I would send the `message.type` as a param in the `useSnackbarStyles` and then decide the final background color inside the styles file: `useSnackbarStyles({ type, backgroundColor })`.

But now that we have 3 clear snackbar states: `info`, `success`, `error` maybe we can remove this `backgroundColor` and just decide it using the `message.type` prop.

Reply:
> But now that we have 3 clear snackbar states: `info`, `success`, `error`

Well, we currently have 2 states: `info` and `error`.

> maybe we can remove this `backgroundColor` and just decide it using the `message.type` prop.

This has some implications. Currently, we are showing a red/green snackbar on the connection lost/restored events. These are not errors, but info messages. If we proceed with your proposal of inferring the color from the type of the message, these snackbars will become white (or whatever bkg color we decide for `info` messages).

I agree with the change though, should I proceed?

Reply:
Yep, but the idea here would be to introduce the `success` state to be able to have the green snackbar as well.

Reply:
That is not the problem. The problem is that currently the `error` state is designed to process real errors and a connection lost event is not an error. How do you comply with this interface?

```
{
      type: "error";
      text?: string;
      raw: unknown;
      parsed: string;
}

Reply:
That's why I would like to have an `ErrorDetails` domain entity with all that fields:

```ts
{
  text?: string;
  raw: unknown;
  parsed: string;
}
```

Because it's an Error in the application (network, decoding...) but it has nothing to do with a UI error. The snackbar interface should be something like this:

```
{
  type: "error"; // red
  error: "string" | ErrorDetails;
}
```

The connection lost is an `Error` if we talk about UI, and the snackbar should be able to display it. We need to decouple domain from views always!

(The names of the interfaces, props etc are just an idea)

Reply:
We agreed to go this way to separate "colored messages" from "actual errors" that can be reported:

```
// User reporting
export type Message =
  | {
      type: "info-msg" | "success-msg" | "error-msg"; // white | green | red
      text: string;
    }
  | {
      type: "error";
      text?: string;
      raw: unknown;
      parsed: string;
    };
```

Reply:
Done!